### DISCUSSION TITLE
Does the training image or instances incresases with the default  augumentation parameters.?

### DISCUSSION THREAD

**Discussion by naveenvj25** [ultralytics/ultralytics#9176] [Q&A]
There are some default parameters related to augmentation. Does it increase the training images or just apply the augmentation to the actual image, like doing some preprocessing?

**Comment by pderrenger** [comment]
Hi there! 👋 Great question! The default augmentation parameters in YOLOv8 apply augmentations to the actual images during training. These operations are like preprocessing steps and do not physically increase the number of images in your dataset. Instead, augmentations like rotations, scaling, and flipping are applied on-the-fly to diversify the training data, helping improve model robustness without needing more images.

For example, during training:

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.yaml')

# Train the model with default augmentations
results = model.train(data='coco128.yaml', epochs=100, imgsz=640)
```

This will utilize default augmentations on your images dynamically each epoch! Hope this clears it up for you! 😊

**Reply by naveenvj25** [reply]
Hi there! Does Augmentations like rotations, scaling, and flipping are applied in training? If so it will increse the data right? Then how the labels are managed..!  Also, I have a dataset with multiclass labels, is it possible to train the model with binary detection by combining these classes into one?

**Reply by pderrenger** [reply]
@naveenvj25 hey! 😊 Yes, augmentations like rotations, scaling, and flipping are indeed applied during training. These augmentations are applied on-the-fly, which means they pseudo-increase the variety of the data without actually increasing the number of images in your dataset. The labels for these augmented images are automatically adjusted to match the transformations, ensuring consistency.

Regarding your dataset with multiclass labels, you can absolutely train it for binary detection by considering all classes as one. You'd simply map all your labels to a single class during preprocessing or within your dataset loader. Here's a tiny snippet to give you an idea:

```python
# Example of modifying labels for binary classification
for labels in dataset_labels:
    labels[:, 0] = 0  # Set all class labels to 0
```

Just make sure that during training, your model is configured for binary classification (`classes=1`). Cheers! 🚀

**Comment by naveenvj25** [comment]
@pderrenger Hi..! Thanks for the information. Hence, changing the default augmentation hyperparameters could effectively increase the data without manually labeling augmented data.  The single class updation should be done in the train.py file, right? Also, is there any possibility of modifying the Yolo backbone architecture to finetune my custom data requirements?..

**Reply by pderrenger** [reply]
Hi @naveenvj25! You're correct that adjusting the augmentation hyperparameters can effectively increase the diversity of your training data without needing additional manual labeling. 🎯

For training with a single class, you don't need to modify `train.py`. Instead, set `single_cls=True` in your training command or configuration file to treat all classes as a single class.

Regarding modifying the YOLO backbone, yes, you can customize the architecture to better suit your data. This involves editing the model's YAML configuration file, where you can adjust layers, filters, and other architectural elements to optimize performance for your specific requirements.

If you need further guidance on how to make these changes, feel free to check out the documentation or ask here! Happy training! 🚀

**Reply by naveenvj25** [reply]
@pderrenger Thanks for the suggestions!

**Reply by pderrenger** [reply]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy coding! 😊
